packages/psss-server/src/connections/MeditrakConnection.ts

```diff
@@ -3,10 +3,112 @@
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
+import { generateId } from '@tupaia/database';
+import { convertPeriodStringToDateRange } from '@tupaia/utils';
 import { ApiConnection } from './ApiConnection';
 
 const { MEDITRAK_API_URL = 'http://localhost:8090/v2' } = process.env;
 
+const SURVEY_RESPONSE_FIELDS = <const>['entity.code', 'survey.code', 'submission_time', 'id'];
+const ANSWER_FIELDS = <const>['question.code', 'id'];
+const PSSS_SURVEY_RESPONSE_ANSWER_TYPE = 'Number'; // All survey response answers are 'Number' in PSSS
+
+type SurveyResponseObject = {
+  [key in typeof SURVEY_RESPONSE_FIELDS[number]]: string;
+};
+
+type AnswerObject = {
+  [key in typeof ANSWER_FIELDS[number]]: string;
+};
+
 export class MeditrakConnection extends ApiConnection {
   baseUrl = MEDITRAK_API_URL;
+
+  async findSurveyResponse(surveyCode: string, orgUnitCode: string, period: string) {
+    const [startDate, endDate] = convertPeriodStringToDateRange(period);
+    const results = (await this.get(`surveyResponses/`, {
+      page: '0',
+      pageSize: '1',
+      columns: `["${SURVEY_RESPONSE_FIELDS.join('","')}"]`,
+      filter: JSON.stringify({
+        'survey.code': { comparisonValue: surveyCode, castAs: 'text' },
+        'entity.code': { comparisonValue: orgUnitCode, castAs: 'text' },
+        submission_time: {
+          comparator: 'BETWEEN',
+          comparisonValue: [startDate, endDate],
+          castAs: 'date',
+        },
+      }),
+      sort: '["submission_time DESC"]',
+    })) as SurveyResponseObject[];
+
+    return results.length > 0 ? results[0] : undefined;
+  }
+
+  async updateSurveyResponse(
+    surveyResponse: SurveyResponseObject,
+    answers: Record<string, number>,
+  ) {
+    const existingAnswers = (await this.get(`surveyResponses/${surveyResponse.id}/answers`, {
+      columns: `["${ANSWER_FIELDS.join('","')}"]`,
+    })) as AnswerObject[];
+
+    const newAnswers = existingAnswers.map(existingAnswer => ({
```

Comment:
We might drop an answer to a question that wasn't answered in the first survey response? Perhaps that's an edge case that never comes up in the flow we're using? I.e., if there's an existing survey response, it will have all questions answered?

Reply:
Yeah I think that's an invalid use case for PSSS. All survey responses must contain case data for each syndrome

Reply:
Great - makes things easy :-)